vignettes/lifecycle.Rmd

```diff
@@ -18,117 +18,101 @@ knitr::opts_chunk$set(
+A function is considered stable when the author is happy with its interface and behaviour and major changes are unlikely.
+Breaking changes will occur gradually, through a deprecation process.
 
-The probable end state of most functions is *stable*:
+The exception to the rule is sub-1.0.0 packages where all functions are considered ![Experimental](figures/lifecycle-experimental.svg){style="vertical-align:middle"}.
+Experimental functions are made available so people can try them out and provide feedback, but the author makes no promises to avoid breaking changes.
+All functions in sub-1.0.0 packages are experimental, but \>=1.0.0 packages can also contain some experimental functions.
 
--   <img src="figures/lifecycle-stable.svg" alt="Stable" style="vertical-align:middle"/> A feature is considered stable when the author is happy with its interface and behaviour. Major changes are unlikely, and breaking changes will occur gradually, through a deprecation process.
+(![Maturing](figures/lifecycle-maturing.svg){style="vertical-align:middle"} functions lie somewhere in between experimental and stable. It's hard to be precise about exactly what this means, so we are gradually phasing out our use of this badge.)
 
-Sometimes we are no longer certain that a feature follows or implements the right approach.
-In this case, we mark it as *questioning*.
-It can then either become *superseded* once we have implemented an alternative, or *deprecated*.
-The difference between superseded and deprecated is that a superseded feature is maintained indefinitely for backward compatibility.
-Alternatively, we may realise that a questioning feature is actually important, and move it back to stable.
-User feedback is usually important for taking these decisions and always welcome.
+### End of life
 
-1.  <img src="figures/lifecycle-questioning.svg" alt="Questioning" style="vertical-align:middle"/> The author is no longer convinced that the feature is the optimal approach.
-    However, there are no recommended alternatives yet.
+There are two ways that a function might reach the end of its life: it might be superseded or deprecated.
 
-2.  <img src="figures/lifecycle-superseded.svg" alt="Superseded" style="vertical-align:middle"/> (Previously called retired) A superseded feature is no longer under active development, and a known better alternative is available.
-    However, it is indefinitely kept in the package for backward compatibility.
-    The author will only make the necessary changes to ensure that the function continues working.
-    No new features will be added, and only critical bugs will be fixed.
+-   A ![Superseded](figures/lifecycle-superseded.svg)[^2] function has a better alternative available, but the function itself is not going away. While a superseded function will not receive new features, it will be kept working for the foreseeable future so that existing code does not need to change.
```

Comment:
A superseded feature thus preserves all the guarantees of a stable feature, and is just as safe to use.